src/database/message-table.c

```diff
@@ -482,25 +482,53 @@ static void format_subnet_message(char *plain, const int sizeof_plain, char *htm
 
 static void format_hostname_message(char *plain, const int sizeof_plain, char *html, const int sizeof_html, const char *ip, const char *name, const int pos)
 {
-	if(snprintf(plain, sizeof_plain, "Host name of client \"%s\" => \"%s\" contains (at least) one invalid character at position %i",
-			ip, name, pos) > sizeof_plain)
+	cJSON *json_name = cJSON_CreateStringReference(name);
+	if(json_name == NULL)
+	{
+		log_err("format_hostname_message(): Failed to create JSON string reference for host name \"%s\" of client \"%s\"", name, ip);
+		return;
+	}
+	const char *namep = cJSON_PrintUnformatted(json_name);
+	cJSON_Delete(json_name);
```

Comment:
We can safely delete the object here, `cJSON_PrintUnformatted()` made a separate string allocation we `free` below.